src/neo/SmartContract/ApplicationEngine.cs

```diff
@@ -55,12 +55,21 @@ internal bool AddGas(long gas)
             return testMode || GasConsumed <= gas_amount;
         }
 
+        internal void RaiseLoadContext(ExecutionContext context)
```

Comment:
After some thinking, I found that `PointerCallback` is actually a call to `System.Contract.Call`, which can also be regarded as a `SyscallCallback` in essence. In this way, we can unify these two callbacks, and then move the calling logic to the `ApplicationEngine`.

Reply:
But this will detach from the current context, that's means that we don't have the same static variables. And `System.Contract.Call` require a public method, not private or internal.

Reply:
I know how to optimize it now. Give me some time.